### PR TITLE
Added some config options and fixed preprocessing

### PR DESCRIPTION
**Pull request by davekiss**
These commits helped with my usage of the plugin and generally increase the flexibility of the plugin. Let me know what you think!

### CODE REVIEW

**Comment by sreuter** [comment]
@davekiss Hey buddy - unfortunately, It's kinda hard to follow the changes because of the additional chore being done. Can you do a quick check if https://github.com/23/resumable.js/pull/232 and https://github.com/23/resumable.js/pull/233 does the job for what you wanted? Thanks!

**Comment by adavidof** [comment]
@steffentchr 
Hello, what about this pull-request?
It will be cool if you add the testTarget functionality

Thank you!

**Comment by steffentchr** [comment]
Sorry -- lost track of this entirely. Problem is that it's a pretty old change set, so hard to merge well. Would you want to do a clean PR; or is someone doing this at the moment?

**Comment by adavidof** [comment]
@steffentchr 
I can do clean PR

**Comment by steffentchr** [comment]
You're nice -- thanks man.

**Comment by adavidof** [comment]
@steffentchr 
Look at this PR, please
https://github.com/23/resumable.js/pull/355
